### ISSUE TITLE
Uses ome layer like batch_normalization when extend tensorflow with tflearn

### ISSUE THREAD

**Issue by shixiazuike** [tflearn/tflearn#398]
When using keras as a simplified interface to tensorflow, some Keras layers (stateful RNNs and BatchNormalization layers) have internal updates that need to be run as part of each training step. There are stored as a list of tensor tuples, layer.updates. We should generate assign ops for those, to be run at each training step. Here's an example:
```python
from keras.layers import BatchNormalization

layer = BatchNormalization()(x)

update_ops = []
for old_value, new_value in layer.updates:
    update_ops.append(tf.assign(old_value, new_value))
```
Do these update_ops needed explictily when extend tensorflow with tflearn? Like this：
```python
import tflearn
import tensorflow as tf 

x = tf.placeholder(dtype, input_shape)
net = tflearn.batch_normalization(x)
...
loss = getattr(tflearn, loss_function)(self.y_pred, self.y_true)
optimizer = tflearn.optimizers.get(optimizer_method)(learning_rate)
optimizer = optimizer.get_tensor()
train_op = optimizer.minimize(loss)

sess = tf.Session()
tflearn.is_training(True, session = sess)
sess.run([train_op, loss], feed_dict=...)
```
Is this correct?

**Comment by aymericdamien** [comment]
TFLearn is using a variable `is_training` to indicate if the network is on training or predicting mode.
Your example is correct, you just need to use back `tflearn.is_training(False, session = sess)` when you want to predict (so mean/var will not be updated for batch norm).

**Comment by harungunaydin** [comment]
What happens if is_training is not set to be True during Training? And what is the default?
